src/main/java/com/expediagroup/hiveberg/IcebergInputFormat.java

```diff
@@ -77,7 +77,11 @@
     URI location = pathAsURI(job.get(TABLE_LOCATION));
 
     // Set defaults for virtual column
-    currentSnapshotId = table.currentSnapshot().snapshotId();
+    try {
+      currentSnapshotId = table.currentSnapshot().snapshotId();
```

Comment:
Where exactly is the NPE coming from? Generally we prefer doing a `if x == null` over a `try/catch` block. Is there something here you can check for null or some other way to see if what need is present?

Reply:
So it's coming from `table.currentSnapshot().snapshotId();` - we're using the Iceberg API to look for the latest snapshot and it doesn't exist yet. So if I try and assign `currentSnapshotId` the NPE gets thrown because of Iceberg, which is why i've gone with the try/catch

Reply:
Is there some other way we can use the API to check for the presence of a snapshot? What does `currentSnapshot` return? I'm hoping there is some way we can use their API to do the check that doesn't throw a NPE.

Reply:
`currentSnapshot` would return a `Snapshot` object but in this case it would be null for the empty table, so I can run the null check on that instead of calling `.snapshotId` straight away to avoid the try/catch